File: hammer-gui/src/main/java/dev/vini2003/hammer/gui/registry/common/HGUIEvents.java
package dev.vini2003.hammer.gui.registry.common;

import dev.vini2003.hammer.gui.api.common.screen.handler.BaseScreenHandler;
import net.fabricmc.fabric.api.event.lifecycle.v1.ServerTickEvents;

public class HGUIEvents {
	public static void init() {
		ServerTickEvents.END_SERVER_TICK.register(server -> {
			for (var player : server.getPlayerManager().getPlayerList()) {
				if (player.currentScreenHandler instanceof BaseScreenHandler screenHandler) {
					screenHandler.tick();
				}
			}
		});
	}
}

File: hammer-gui/src/main/kotlin/dev/vini2003/hammer/gui/registry/common/HGUIEvents.kt
package dev.vini2003.hammer.gui.registry.common

import dev.vini2003.hammer.gui.common.listener.ScreenHandlerServerTickListener
import net.fabricmc.fabric.api.event.lifecycle.v1.ServerTickEvents

object HGUIEvents {
	fun init() {
		ServerTickEvents.END_SERVER_TICK.register(ScreenHandlerServerTickListener)
	}
}